cricPredict/Contest/views.py

```diff
@@ -1,3 +1,239 @@
-from django.shortcuts import render
+import datetime
 
 # Create your views here.
+from django.db.models import Sum
+from rest_framework import status
+from rest_framework.exceptions import ParseError
+from rest_framework.generics import GenericAPIView, get_object_or_404
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from Contest.models import League, Match, Prediction
+from Contest.serializers import LeagueSerializer, MatchSerializer, PredictionSerializer, \
+    ExtendedMatchSerializer, ExtendedPredictionSerializer, ExtendedLeagueSerializer, RankingsSerializer
+from Groups.models import Group
+
+
+class LeagueAPIView(APIView):
+    permissions = (IsAuthenticated,)
+    serializer_class = LeagueSerializer
+    queryset = ''
+
+    def post(self, request, *args, **kwargs):
+        serializer = LeagueSerializer(data=self.request.data)
```

Comment:
we can always use the `ListCreateAPIView` and `RetrieveUpdateDestroyAPIView` to cleanup these views